packages/smooth_app/lib/pages/product/common/product_list_helper.dart

```diff
@@ -0,0 +1,101 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/data_models/product_list.dart';
+import 'package:smooth_app/database/dao_product.dart';
+import 'package:smooth_app/database/dao_product_list.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/pages/product/common/product_refresher.dart';
+
+class ProductListHelper {
+  const ProductListHelper({required this.list, required this.localDatabase});
+
+  final ProductList list;
+  final LocalDatabase localDatabase;
+
+  static const String EXPORT_SEPARATOR = ';';
+
+  Future<String> exportBarcodesToString() async {
+    final StringBuffer csv = StringBuffer();
+
+    void addLine(String barcode, String name, String brand) {
+      csv.writeln(
+        '$barcode'
+        '$EXPORT_SEPARATOR'
+        '$name'
+        '$EXPORT_SEPARATOR'
+        '$brand',
+      );
+    }
+
+    addLine('Barcode', 'Name', 'Brand');
+
+    final List<String> validBarcodes = list.barcodes.where(
+      (String barcode) {
+        return isBarcodeValid(barcode);
+      },
+    ).toList();
```

Comment:
Still puzzled by the added value of that "barcode validation".
I would just send whatever the user asked for: all the barcodes of the list.